### ISSUE TITLE
Bicep generated old apiVersions for "Microsoft.Resources/deployments" caused the failure of test "apiVersions Should Be Recent In Reference Functions"

### ISSUE THREAD

**Issue by majguo** [Azure/arm-ttk#704]
This is another issue relates to the issue which has been resolved by #699.

Running arm-ttk test against an ARM template generated by Bicep, it complains:

```
[-] apiVersions Should Be Recent In Reference Functions (611 ms)
        Api versions must be the latest or under 2 years old (730 days) - API version used by:
            reference(resourceId('Microsoft.Resources/deployments', 'query-available-private-ip-for-app-gateway'), '2020-10-01')
        is 742 days old Line: 673, Column: 60
        Valid Api Versions for Microsoft.Resources/deployments :
        2021-04-01
        2021-04-01
        2021-01-01
```

Similar to the fix of #699, maybe the apiVersion check for `Microsoft.Resources/deployments` can also be skipped in test `apiVersions Should Be Recent In Reference Functions`?

## How to reproduce

1. Install [Bicep](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/install#linux)
1. Checkout the latest [arm-ttk](https://github.com/Azure/arm-ttk)
1. Checkout [azure.liberty.aks](https://github.com/majguo/azure.liberty.aks) where the issue was observed
1. Run the following commands to build ARM template and run arm-ttk tests:
    ```
    cd <path-to-your-clone-of-azure.liberty.aks>/src/main/bicep
    bicep build mainTemplate.bicep
    <path-to-your-clone-of-arm-ttk>/arm-ttk/Test-AzTemplate.sh -TemplatePath ./mainTemplate.json
    ```

**Comment by edburns** [comment]
I have mentioned this in the official tracking issue for all things coming from old apiVersions in bicep.

https://github.com/Azure/bicep/issues/7932

**Comment by bmoore-msft** [comment]
s/b fixed in #705
